pkg/cloudprovider/providers/vsphere/vclib/diskmanagers/vdm.go

```diff
@@ -57,15 +57,20 @@ func (diskManager virtualDiskManager) Create(ctx context.Context, datastore *vcl
 	if err != nil {
 		vclib.RecordvSphereMetric(vclib.APICreateVolume, requestTime, err)
 		glog.Errorf("Failed to create virtual disk: %s. err: %+v", diskManager.diskPath, err)
-		return err
+		return "", err
 	}
-	err = task.Wait(ctx)
+	taskInfo, err := task.WaitForResult(ctx, nil)
+	if err != nil {
+		glog.Errorf("Failed to complete virtual disk creation: %s. err: %+v", diskManager.diskPath, err)
+		return "", err
+	}
+	canonicalDiskPath = taskInfo.Result.(string)
 	vclib.RecordvSphereMetric(vclib.APICreateVolume, requestTime, err)
```

Comment:
when it comes to here, even if RecordvSphereMetric failed, does it mean the disk is not created or the disk could be already create? Do we need to care about clean up the disk if it fails here?

Reply:
RecordvSphereMetric would never fail/throw exception. 
However, I have made a small correction here to record vSphere metric irrespective of the error. This will be useful for the admin.